tpl_lib/dxf/dxf.tpl

```diff
@@ -383,49 +446,52 @@ module.exports = extend({
   },
 
 
-  polyline_cut: function(pl) {
+  polyline_cut: function(pl, zCut) {
+    var v = first(this.element_vertices(pl));
+    cut({z: zCut});
+    cut(v.x, v.y);
+
     for (var i = 1; i < pl.vertices.length; i++)
       cut(pl.vertices[i].x, pl.vertices[i].y);
 
     if (pl.vertices.length) cut(pl.vertices[0].x, pl.vertices[0].y);
   },
 
 
-  spline_cut: function(s, res) {
+  spline_cut: function(s, zCut, res) {
     if (typeof res == 'undefined') res = units() == METRIC ? 1 : 1 / 25.4;
 
-    if (s.degree == 2) {
-      var steps = Math.ceil(quad_bezier_length(s.ctrlPts) / res);
-      var delta = 1 / steps;
-
-      for (var i = 0; i < steps; i++) {
-        v = quad_bezier(s.ctrlPts, delta * (i + 1));
-        cut(v.x, v.y);
-      }
+    var steps = Math.ceil(nurbs_length(s) / res);
+    //var delta = 1.0 / steps;
+    var delta = 0.01;
```

Comment:
Why not use the value from ``nurbs_length()`` here?

Reply:
because of the bug you found at line 238 :-D
sorry for pushing in a rush, but  that was it for my 30 minutes of coding.
I have one free hour now, I'll address all comments and try to implement closed curves at least.